apps/minifront/src/state/swap/helpers.ts

```diff
@@ -63,22 +64,69 @@ export const sendCandlestickDataRequest = async (
     throw new Error('Asset pair equivalent');
   }
 
-  try {
-    const { data } = await dexClient.candlestickData(
+  const suppressAbort = (err: unknown) => {
+    if (signal && !signal.aborted) {
+      throw err;
+    }
+  };
+
+  const directReq = dexClient
+    .candlestickData(
       {
         pair: { start, end },
         limit,
       },
       { signal },
-    );
-    return data;
-  } catch (err) {
-    if (err instanceof Error && err.name === 'AbortError') {
-      return;
-    } else {
-      throw err;
+    )
+    .catch(suppressAbort);
+
+  const inverseReq = dexClient
+    .candlestickData(
+      {
+        pair: { start: end, end: start },
+        limit,
+      },
+      { signal },
+    )
+    .catch(suppressAbort);
+
+  const candlestickData = (await directReq)?.data ?? [];
+
+  const inverseCandlestickData = (await inverseReq)?.data ?? [];
+
+  const collated = Map.groupBy(
+    [
+      ...candlestickData,
+      ...inverseCandlestickData.map(
+        cd =>
+          new CandlestickData({
+            ...toPlainMessage(cd),
+            close: 1 / cd.close,
+            high: 1 / cd.low,
+            low: 1 / cd.high,
+            open: 1 / cd.open,
+          }),
+      ),
+    ],
+    ({ height }) => height,
+  );
+
+  const combined = Array.from(collated.values()).map(([first, extra]) => {
```

Comment:
think it would be clearer to stick with `direct` & `inverse` versus `first` & `extra`

Reply:
order and presence of either are not guaranteed. this was refactored to use 'reduce'